src/rdata/zonemd.rs

```diff
@@ -0,0 +1,447 @@
+where
+    Octs: AsRef<[u8]> + ?Sized,
+    Other: AsRef<[u8]> + ?Sized,
+{
+    fn eq(&self, other: &Zonemd<Other>) -> bool {
+        self.serial.eq(&other.serial)
+            && self.scheme.eq(&other.scheme)
+            && self.algo.eq(&other.algo)
+            && self.digest.as_ref().eq(other.digest.as_ref())
+    }
+}
+
+impl<Octs: AsRef<[u8]> + ?Sized> Eq for Zonemd<Octs> {}
+
+// section 2.4
+impl<Octs: AsRef<[u8]>> fmt::Display for Zonemd<Octs> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{} {} {} ( ",
+            self.serial,
+            u8::from(self.scheme),
+            u8::from(self.algo)
+        )?;
+        base16::display(&self.digest, f)?;
+        write!(f, " )")
+    }
+}
+
+impl<Octs: AsRef<[u8]>> fmt::Debug for Zonemd<Octs> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str("Zonemd(")?;
+        fmt::Display::fmt(self, f)?;
+        f.write_str(")")
+    }
+}
+
+impl<Octs, Other> PartialOrd<Zonemd<Other>> for Zonemd<Octs>
+where
+    Octs: AsRef<[u8]>,
+    Other: AsRef<[u8]>,
+{
+    fn partial_cmp(&self, other: &Zonemd<Other>) -> Option<Ordering> {
+        match self.serial.partial_cmp(&other.serial) {
+            Some(Ordering::Equal) => {}
+            other => return other,
+        }
+        match self.scheme.partial_cmp(&other.scheme) {
+            Some(Ordering::Equal) => {}
+            other => return other,
+        }
+        match self.algo.partial_cmp(&other.algo) {
+            Some(Ordering::Equal) => {}
+            other => return other,
+        }
+        self.digest.as_ref().partial_cmp(other.digest.as_ref())
+    }
+}
+
+impl<Octs, Other> CanonicalOrd<Zonemd<Other>> for Zonemd<Octs>
+where
+    Octs: AsRef<[u8]>,
+    Other: AsRef<[u8]>,
+{
+    fn canonical_cmp(&self, other: &Zonemd<Other>) -> Ordering {
+        match self.serial.into_int().cmp(&other.serial.into_int()) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.scheme.cmp(&other.scheme) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.algo.cmp(&other.algo) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        self.digest.as_ref().cmp(other.digest.as_ref())
+    }
+}
+
+impl<Octs: AsRef<[u8]>> Ord for Zonemd<Octs> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.serial.into_int().cmp(&other.serial.into_int()) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.scheme.cmp(&other.scheme) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.algo.cmp(&other.algo) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        self.digest.as_ref().cmp(other.digest.as_ref())
+    }
+}
+
+/// The Scheme is an 8-bit unsigned integer that identifies the
```

Comment:
Could you start the docstring with a single one-line summary? This will then show in the index pages of the module and gives a hint as to what the type is for …